contacts/main.js

```diff
@@ -0,0 +1,566 @@
+    id: id,
+    firstName: 'Philip',
+    lastName: 'Pope',
+    image: 'images/profile-pics/pic5.jfif',
+    likes: 0
+  };
+  person = new Person(obj);
+  array.push(person);
+  id = generateUniqueID();
+  obj = {
+    id: id,
+    firstName: 'Viorel',
+    lastName: 'Popescu',
+    image: 'images/profile-pics/pic6.jfif',
+    likes: 0
+  };
+  person = new Person(obj);
+  array.push(person);
+  id = generateUniqueID();
+  obj = {
+    id: id,
+    firstName: 'Elon',
+    lastName: 'Musk',
+    image: 'images/profile-pics/pic7.jfif',
+    likes: 0
+  };
+  person = new Person(obj);
+  array.push(person);
+
+  return array;
+}
+
+function saveDataToLocalStorage() {
+  localStorage.setItem('contacts', JSON.stringify(contactsList));
+}
+
+function getContactsFromLocalStorage() {
+  return JSON.parse(localStorage.getItem('contacts'));
+}
+
+function populateContactsListHTML() {
+  for (let i = 0; i < contactsList.length; i++) {
+    addToHTMLNewPerson(contactsList[i]);
+  }
+}
+
+function consoleLogContactsList() {
+  console.log('-----------------------------------------------------');
+  for (let i = 0; i < contactsList.length; i++) {
+    console.log(contactsList[i].getFullNameAndLikes());
+  }
+}
+
+function refreshFavsList() {
+  // clear list
+  document.getElementById('favs-list').innerHTML = '';
+  // calculate the new list 
+  let favsMap = calculateFavsList();
+  // rerender the new list
+  rerenderFavsListHTML(favsMap);
+}
+
+function calculateFavsList() {
+  // returns a map
+  let mp = {};
+  // we deep copy it
+  let arr = [...contactsList];
+  // we sort it by likes
+  sortArrayByLikes(arr);
+  // we create the map of arrays
+  mp = createMapOfFavs(arr);
+  // sort each of them
+  Object.keys(mp).forEach((key) => {
+    sortArrayAlphabetically(mp[key]);
+  })
+  return mp;
+}
+
+function sortArrayByLikes(arrayOfContacts) {
+  // arrayOfContacts - array of objects person
+  arrayOfContacts.sort((p1, p2) => {
+    return p2.likes - p1.likes;
+  })
+}
+
+function createMapOfFavs(arrayOfContacts) {
+  let myMap = {};
+  for (let i = 0; i < arrayOfContacts.length; i++) {
+    let maxim = arrayOfContacts[i].likes;
+    if (typeof myMap[maxim] === 'undefined') {
+      // we create a new one
+      myMap[maxim] = [];
+      myMap[maxim].push(arrayOfContacts[i]);
+    } else {
+      myMap[maxim].push(arrayOfContacts[i]);
+    }
+  }
+  return myMap;
+}
+
+function sortArrayAlphabetically(arrayOfContacts) {
+  arrayOfContacts.sort((p1, p2) => {
+    let p1FullName = p1.firstName + ' ' + p1.lastName;
+    let p2FullName = p2.firstName + ' ' + p2.lastName;
+    if (p1FullName > p2FullName) return 1;
+    if (p1FullName < p2FullName) return -1;
+    return 0;
+  })
+}
+
+function rerenderFavsListHTML(mapOfContacts) {
+
+  let ol = document.getElementById('favs-list');
+  ol.innerHTML = '';
+
+  let numbers = Object.keys(mapOfContacts).sort();
+  numbers.reverse();
+
+  for (let i = 0; i < numbers.length; i++) {
+    if (i === 3) break;
```

Comment:
usually we prefer extracting numbers to constants, so that we know what they refer to

const NUMBER_OF_FAVS = 3